src/test/java/at/fhv/ss22/ea/f/musicshop/backend/unit/application/ProductApplicationTests.java

```diff
@@ -150,4 +151,152 @@ void given_product_artist_and_carrier_when_get_product_by_id_for_then_dto_values
+                            Song.create("Spring"+i, "3:00"+i),
+                            Song.create("Wo bist du"+i, "3:00"+i),
+                            Song.create("Stirb nicht vor mir!"+i, "3:00"+i)
+                    )
+            );
+            products.add(p);
+        }
+
+        List<ProductId> productIds = new ArrayList<>();
+
+        for(int i = 0; i < products.size(); i++){
+            productIds.add(products.get(i).getProductId());
+        }
+
+        Artist rammstein = Artist.create(
+                rammsteinId,
+                "rammstein",
+                "deutschland",
+                productIds
+        );
+        when(mockedProductRepository.fullTextSearch(anyString())).thenReturn(products);
+        when(mockedArtistRepo.artistById(rammsteinId)).thenReturn(Optional.of(rammstein));
+
+        //when
+        List<ProductOverviewDTO> productDTOs = productApplicationService.search("irrelevant to this test", 0);
+
+        //then
+        assertEquals(products.size(), productDTOs.size());
+    }
+
+    @Test
+    void when_page_number_equals_two_and_two_exists_then_return_two() throws SessionExpired, NoPermissionForOperation {
+        //given
+        ArtistId rammsteinId = new ArtistId(UUID.randomUUID());
+
+        List<Product> products = new ArrayList<>();
+
+        for(int i = 1; i <= 100; i++){
+            Product p = Product.create(
+                    new ProductId(UUID.randomUUID()),
+                    "Rosenrot"+i,
+                    "2000"+i,
+                    List.of("Rock", "Metal"+i),
+                    "Rammstein GBR"+i,
+                    "40:00"+i,
+                    List.of(rammsteinId),
+                    List.of(
+                            Song.create("Benzin"+i, "3:46"+i),
+                            Song.create("Mann gegen Mann"+i, "3:00"+i),
+                            Song.create("Rosenrot"+i, "3:00"+i),
+                            Song.create("Spring"+i, "3:00"+i),
+                            Song.create("Wo bist du"+i, "3:00"+i),
+                            Song.create("Stirb nicht vor mir!"+i, "3:00"+i)
+                    )
+            );
+            products.add(p);
+        }
+
+        List<ProductId> productIds = new ArrayList<>();
+
+        for(int i = 0; i < products.size(); i++){
+            productIds.add(products.get(i).getProductId());
+        }
+
+        Artist rammstein = Artist.create(
+                rammsteinId,
+                "rammstein",
+                "deutschland",
+                productIds
+        );
+        when(mockedProductRepository.fullTextSearch(anyString())).thenReturn(products);
+        when(mockedArtistRepo.artistById(rammsteinId)).thenReturn(Optional.of(rammstein));
+
+        //when
+        List<ProductOverviewDTO> productDTOs = productApplicationService.search("irrelevant to this test", 2);
+        //then
+        assertEquals(20, productDTOs.size());
```

Comment:
i würd do die 20 vorher bei given in a Variable expectedSize tua und selber die page 2 als liste macha zum denn oh abprüfa ob vo da elemente her die gliche liste zruck kummt.